hooks/utils.py

```diff
@@ -322,17 +323,21 @@ def cmd_log(results):
 
 
 def write_gui_config(
-        console_enabled, login_help, readonly, charmworld_url, charmstore_url,
-        build_dir, secure=True, sandbox=False, cached_fonts=False,
-        hide_login_button=False, config_js_path=None, ga_key='',
-        juju_core_version=None, password=None, juju_env_uuid=None):
+        sandbox=False, debug=False, juju_env_user=None, juju_env_password=None):
     """Generate the GUI configuration file."""
 
     log('Generating the Juju GUI configuration file to {}.'.format(INI_PATH))
-    ini = read_config()
+    if os.path.exists(INI_PATH):
+        ini = read_config(INI_PATH)
+    else:
+        ini = read_config(INI_TEMPLATE_PATH)
```

Comment:
This might cause trouble on upgrade-charm.  We'd never get new additions to the template.